src/box/lua/net_box.c

```diff
@@ -2043,13 +2057,15 @@ netbox_console_setup(struct lua_State *L)
 	int fd = lua_tointeger(L, 1);
 	struct ibuf *send_buf = (struct ibuf *)lua_topointer(L, 2);
 	struct ibuf *recv_buf = (struct ibuf *)lua_topointer(L, 3);
+	struct fiber_cond *send_cond =
+		(struct fiber_cond *)luaL_checkudata(L, 4, cond_typename);
```

Comment:
netbox_iproto_auth requires condvar signaling, as well as console, to keep the close () code simpler. Otherwise in close, we must check that we have state == 'active' or state == 'fetch_schema', as well as that we are not using the console (otherwise, upon shutdown, we risk hang).

Reply:
Agree.